tracer/src/Datadog.Tracer.Native/debugger_method_rewriter.cpp

```diff
@@ -1342,6 +1372,37 @@ HRESULT DebuggerMethodRewriter::Rewrite(RejitHandlerModule* moduleHandler,
         return S_FALSE; // TODO https://datadoghq.atlassian.net/browse/DEBUG-706
     }
 
+    if (!isAsyncMethod)
+    {
+        // Async methods can't have Task<T> where T is byref like, because byref like can't exist as a generic param.
+        // thus we only perform this check here.
```

Comment:
Not sure I follow the logic behind this comment - why are we performing the check then?

Reply:
I'm not sure what you mean. In async methods, the return value can not be byref-like so I'm only performing that check on non-async methods.

Reply:
That's a clearer explanation :) proposed to use that as a comment in https://github.com/DataDog/dd-trace-dotnet/pull/3505/files/2534a591397357779c4f204d1b08a663c8f513ac#r1031768826, resolving this one.